dm/worker/server.go

```diff
@@ -59,12 +59,15 @@ var (
 // sends responses to RPC client.
 type Server struct {
 	// closeMu is used to sync Start/Close and protect 5 fields below
-	closeMu    sync.Mutex
-	closed     atomic.Bool
-	inited     bool
-	rootLis    net.Listener
-	svr        *grpc.Server
-	etcdClient *clientv3.Client
+	closeMu sync.Mutex
+	//closed is used to store the current state of the dm-worker server.
+	//if the server starts successfully, s.closed.Store(false)
+	closed atomic.Bool
+	//calledClose is used to indicate that dm-worker has received signal to close and closed successfully.
```

Comment:
```suggestion
	// calledClose is used to indicate that dm-worker has received signal to close and closed successfully.
	// we use this variable to avoid Start() after Close()
```